libheif/plugins/encoder_aom.cc

```diff
@@ -373,6 +375,14 @@ static void aom_init_parameters()
   p->has_default = true;
   d[i++] = p++;
 
+  assert(i < MAX_NPARAMETERS);
```

Comment:
Should we add a
`#if defined(AOM_CTRL_AV1E_SET_ENABLE_INTRABC)`
around this block such that the option is only available when supported by our AOM version?

Reply:
Good question. None of the other parameters currently do this, and many of those are only available in more recent versions of libaom. I guess we should probably be consistent either way. Adding these guards might make the (debug only) `MAX_NPARAMETERS` checking more complex too.

Reply:
Then we should add such a `#if` to the other parameters too unless they are so old that we can assume that they will always be present.

The `assert(i < MAX_NPARAMETERS)` doesn't have to be modified. It is just a quick and dirty way to make sure that we do not write beyond the boundaries of the fixed-size array.

Reply:
Thanks Dirk, I've added an `#if` wrapper around the use of this parameter and left existing code as-is for now.

Reply:
I looked into this. `AOM_CTRL_AV1E_SET_ENABLE_INTRABC` first appeared in the `v1.0.0-errata1-avif` tag. (That tag was created for the libavif library, so libaom versions older that that tag don't need to be supported by an AVIF library.)  We can assume `AOM_CTRL_AV1E_SET_ENABLE_INTRABC` is defined and remove the `#if defined(AOM_CTRL_AV1E_SET_ENABLE_INTRABC)` checks.

Reply:
@wantehchang I don't understand the reasoning. Why couldn't an aom version before `v1.0.0-errata1-avif` be used together with libheif v1.19?

However, given that the intra-bc flag was added in Dec. 2019, it is probably safe to assume it exists.

Reply:
Dirk: I could not find libheif's minimum supported version of libaom. That's why I used the `v1.0.0-errata1-avif` tag, which was created for libavif (presumably the first version of libaom ever supported by libavif), to argue that any other AVIF library is likely to need a version of libaom >= `v1.0.0-errata1-avif`.

In my opinion there is no need to support libaom versions < v3.0.0. The latest libaom release is v3.11.0.